### PR TITLE
TEST-#4363: Use Ray from pypi in CI

### LINKED ISSUE #4363: Ray Segfaulting on Windows in CI

### PR DESCRIPTION
**Pull request by devin-petersohn**
Signed-off-by: Devin Petersohn <devin.petersohn@gmail.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4363  
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Comment by YarShev** [comment]
@devin-petersohn, there is still a seg fault in CI. Should we pip Ray to a lower version?

**Comment by vnlitvinov** [comment]
I wonder how that helps - it effectively uses Ray 1.10, right?

**Comment by devin-petersohn** [comment]
@vnlitvinov Not sure what is happening with conda to cause this seg fault, but this tests with Ray 1.11

I am not able to reproduce the seg faults locally

File: environment-dev.yml [hunk 1]
```diff
@@ -43,3 +42,4 @@ dependencies:
       - modin-spreadsheet>=0.1.1
       - tqdm
       - git+https://github.com/airspeed-velocity/asv.git@ef016e233cb9a0b19d517135104f49e0a3c380e9
+      - ray
```
Comment (YarShev) [inline environment-dev.yml:45]: We install `ray[default]` in requirements-dev.txt and should probably do the same here.

**Comment by devin-petersohn** [comment]
@vnlitvinov That line was removed in the latest commits.

File: environment-dev.yml (hunk 1 above)
Comment (devin-petersohn) [reply environment-dev.yml]: Done!

File: environment-dev.yml [hunk 2]
```diff
@@ -43,3 +42,4 @@ dependencies:
       - modin-spreadsheet>=0.1.1
       - tqdm
       - git+https://github.com/airspeed-velocity/asv.git@ef016e233cb9a0b19d517135104f49e0a3c380e9
+      - ray[default]
```
Comment (YarShev) [inline environment-dev.yml:45]:
```suggestion
      - ray[default]>=1.4.0
```

**Review by YarShev — APPROVED** [review]
Will merge when CI passes.

File: docs/release_notes/release_notes-0.15.0.rst [hunk 3]
```diff
@@ -20,7 +20,7 @@ Key Features and Updates
 * Developer API enhancements
   *
 * Update testing suite
-  *
+  * Use Ray from pypi in CI (#4364)
```
Comment (YarShev) [inline docs/release_notes/release_notes-0.15.0.rst:23]:
```suggestion
  * TEST-#4363: Use Ray from pypi in CI (#4364)
```

**Pull request merged**